apps/meteor/server/methods/registerUser.ts

```diff
@@ -129,6 +129,12 @@ Meteor.methods<ServerMethods>({
 			// throw new Meteor.Error 'error-email-send-failed', 'Error trying to send email: ' + error.message, { method: 'registerUser', message: error.message }
 		}
 
+		const defaultRoles = settings.get('Accounts_Registration_Users_Default_Roles');
+
+		if (!defaultRoles) {
+			await Users.updateOne({ _id: userId }, { $set: { roles: [] } });
+		}
+
```

Comment:
```suggestion
```

Do not handle this here, but on the same place where the roles are loaded today:

https://github.com/RocketChat/Rocket.Chat/blob/dce4a829fa63fcb55ede1bd49bbe0b78cb927af9/apps/meteor/app/authentication/server/startup/index.js#L253

And instead of using an empty array, add `user` to the role list if the list is empty.

Reply:
> And instead of using an empty array, add user to the role list if the list is empty.
but that goes against the intention, what iF X wants to not set any default role?

Reply:
if we want at least one role, and that being "user", it should reflect on the ui as well.

Reply:
Okay, I'll move the logic to the correct place.

> And instead of using an empty array, add user to the role list if the list is empty.

Aren't we forcing some role if the default role is empty? This is the correct behavior to expect from the application?

Reply:
We have several places that force the 'user' role if nothing else is set, but there's actually no reason for that. Leaving it empty is fine.